python/ray/dataframe/utils.py

```diff
@@ -282,3 +282,60 @@ def decorator(cls):
         return cls
 
     return decorator
+
+
+def _map(func, *objects):
+    """Maps a function across a list of objects stored in Ray.
+
+    Args:
+        func (callable): function taking 1 argument.
+        objects: list of objects or ObjectIds.
+
+    Returns:
+        List of objects returned by func.
+    """
+    return [_deploy_func.remote(func, obj) for obj in objects]
+
+
+@ray.remote
+def _reduce(func, *sequence):
+    """Reduces a sequence to a single value.
+
+    Args:
+        func (callable): function mapping two arguments to a single value.
+        sequence (iterable): list of objects or object IDs.
+
+    Returns:
+        Object returned by cumulatively applying func.
+    """
+    size = len(sequence)
+    if size == 1:
```

Comment:
Generally most tree-reduce implementations don't reduce down to one value due to overhead, so it's probably more worth it to place a larger base case size and enforce that the reduction function is compatible with python's builtin `reduce`.